x/ibc-transfer/handler_test.go

```diff
@@ -29,41 +28,38 @@ func (suite *HandlerTestSuite) SetupTest() {
-	handlerA := ibctransfer.NewHandler(suite.chainA.App.TransferKeeper)
 
-	coinToSendToB := sdk.NewCoins(sdk.NewCoin(fmt.Sprintf("%s/%s/%s", channelB.PortID, channelB.ID, sdk.DefaultBondDenom), sdk.NewInt(100)))
+	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
 
 	// send from chainA to chainB
 	msg := types.NewMsgTransfer(channelA.PortID, channelA.ID, coinToSendToB, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), 110, 0)
 
-	res, err := handlerA(suite.chainA.GetContext(), msg)
-	suite.Require().NoError(err)
-	suite.Require().NotNil(res, "%+v", res) // successfully executed
-
-	err = suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, msg)
+	err := suite.coordinator.SendMsgs(suite.chainA, suite.chainB, clientB, msg)
 	suite.Require().NoError(err) // message committed
 
-	handlerB := ibctransfer.NewHandler(suite.chainB.App.TransferKeeper)
+	// relay send
+	fungibleTokenPacket := types.NewFungibleTokenPacketData(coinToSendToB.Denom, coinToSendToB.Amount.Uint64(), suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String())
+	packet := channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, 110, 0)
+	ack := types.FungibleTokenPacketAcknowledgement{Success: true}
+	err = suite.coordinator.RelayPacket(suite.chainA, suite.chainB, clientA, clientB, packet, ack.GetBytes())
+	suite.Require().NoError(err) // relay committed
 
-	coinToSendBackToA := sdk.NewCoins(sdk.NewCoin(fmt.Sprintf("%s/%s/%s", channelA.PortID, channelA.ID, sdk.DefaultBondDenom), sdk.NewInt(100)))
+	coinToSendBackToA := types.GetTransferCoin(channelB.PortID, channelB.ID, sdk.DefaultBondDenom, 100)
 
 	// send from chainB back to chainA
-	msg = types.NewMsgTransfer(channelA.PortID, channelA.ID, coinToSendBackToA, suite.chainB.SenderAccount.GetAddress(), suite.chainA.SenderAccount.GetAddress().String(), 110, 0)
-
-	res, err = handlerB(suite.chainB.GetContext(), msg)
-	suite.Require().NoError(err)
-	suite.Require().NotNil(res, "%+v", res) // successfully executed
+	msg = types.NewMsgTransfer(channelB.PortID, channelB.ID, coinToSendBackToA, suite.chainB.SenderAccount.GetAddress(), suite.chainA.SenderAccount.GetAddress().String(), 110, 0)
 
 	err = suite.coordinator.SendMsgs(suite.chainB, suite.chainA, clientA, msg)
 	suite.Require().NoError(err) // message committed
+
+	// relay send
+	fungibleTokenPacket = types.NewFungibleTokenPacketData(coinToSendBackToA.Denom, coinToSendBackToA.Amount.Uint64(), suite.chainB.SenderAccount.GetAddress().String(), suite.chainA.SenderAccount.GetAddress().String())
+	packet = channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, channelB.PortID, channelB.ID, channelA.PortID, channelA.ID, 110, 0)
+	err = suite.coordinator.RelayPacket(suite.chainB, suite.chainA, clientB, clientA, packet, ack.GetBytes())
+	suite.Require().NoError(err) // relay committed
```

Comment:
Can you test that chainA received the original tokens with correct denom and amount, and that escrow address decreased balance as expected?